libqtile/core/manager.py

```diff
@@ -749,14 +749,17 @@ def cmd_focus_by_click(self, e):
 
         # Additional option for config.py
         # Brings clicked window to front
+        window = self.windows_map.get(wnd)
         if self.config.bring_front_click:
-            self.conn.conn.core.ConfigureWindow(
-                wnd,
-                xcffib.xproto.ConfigWindow.StackMode,
-                [xcffib.xproto.StackMode.Above]
-            )
+            if window:
+                window.to_relative_top()
+            else:
```

Comment:
When do we enter in that block?

Reply:
When `window` is at all managed by qtile. If it isn't, it doesn't matter that we can't sync its z-property with a group and we can just do the raw call to `ConfigureWindow`.

Reply:
Does it make sense that the function works with a window that is not handled by Qtile?

Reply:
`window = self.windows_map.get(wnd)` was already there, I just moved it up a little. I honestly don't know whether a non-managed window makes sense here, but somebody else did.

Reply:
Right! :)